extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1052,6 +1052,14 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /**
+   * Returns the last URL used for an ad request.
+   * @return {?string}
+   */
+  getCachedAdUrl() {
+    return this.adUrl_;
```

Comment:
I don't think you need this function. If you are only accessing getCachedAdUrl from within the doubleclick fast fetch class, you can just directly use this.adUrl_

Reply:
I'd need to make `adUrl_` protected, which would require a lot more changes. Unless you have a strong objection, I'd rather expose it through a getter function.

Reply:
ah k, idk why I was thinking you could do that.

Reply:
Doesn't doubleclick impl know the ad url?  After all it constructed it when it overrode getAdUrl.  You could instead have a local variable that encapsulates the message to be sent (as a typdef) which you would update (e.g. set content url, creative id, line item id, etc).  When in doubt, we should try not to clutter that AmpA4A API unless necessary.